base/headers.go

```diff
@@ -161,15 +173,13 @@ func (uri *SipUri) String() string {
 	}
 
 	// Optional userinfo part.
-	if uri.User != nil {
-		buffer.WriteString(*uri.User)
-
-		if uri.Password != nil {
-			buffer.WriteString(":")
-			buffer.WriteString(*uri.Password)
+	switch user := uri.User.(type) {
+	case String:
+		buffer.WriteString(user.String())
+		switch pw := uri.Password.(type) {
+		case String:
+			buffer.WriteString(pw.String())
```

Comment:
I think you've lost the old line 168 here (buffer.WriteString(":")). If so, and if this didn't cause any failed tests, can you add a test?